staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go

```diff
@@ -129,9 +129,21 @@ func (t *azureFileCSITranslator) TranslateInTreePVToCSI(pv *v1.PersistentVolume)
 			resourceGroup = v
 		}
 	}
+
+	// Secret is required when mounting a volume but pod presence cannot be assumed - we should not try to read pod now.
 	namespace := defaultSecretNamespace
+	// First - try to read SecretNamespace from source pv.
 	if azureSource.SecretNamespace != nil {
 		namespace = *azureSource.SecretNamespace
+	} else {
+		// Second - try to read namespace from ClaimRef which should be always present.
+		if pv.Spec.ClaimRef != nil && pv.Spec.ClaimRef.Namespace != "" {
+			namespace = pv.Spec.ClaimRef.Namespace
+		} else {
```

Comment:
If this should never happen, then I think we should return an error instead of falling back to `defaultSecretNamespace`.

That should be consistent with intree logic: https://github.com/kubernetes/kubernetes/blob/64e83a7e43afd038b3d5a0179e85099a0958de4c/pkg/volume/azure_file/azure_file.go#L380

Reply:
is it possible that `pv.Spec.ClaimRef.Namespace` is empty which means it's in default namespace? if so, add a info logging would be ok.

Reply:
@RomanBednar are you able to remove the warning msg code L142-L146? I think `pv.Spec.ClaimRef.Namespace` could be empty, and then we use `defaultSecretNamespace`

Reply:
@msau42 I did not describe it too well: the default should not happen while mounting the volume, because that will result in failed mount if secret is in different namespace. But when unmounting the secret is not needed so default can be safely used. That means returning an error would break the unmount flow?

Having it consistent would be ideal, but I don't think that's possible here because this translation code can run when a pod is not created yet. Unlike the intree getSecretNameAndNamespace() which seems to be always called with pod.Namespace which is considered the default.

@andyzhangx I'm not sure empty means it's in default namespace, I'll remove the log - it's not meaningful for unmount flows.

Reply:
For the `pv.Spec.ClaimRef.Namespace` empty case, we should follow intree behavior, which would set it as "".

For unmount, if we don't use the secret, can we leave both `secret` and `secretNamespace` empty? 

What I want to make sure, is that we don't reference a secret we're not supposed to, especially across namespaces.

Reply:
the unmount process won't call `TranslateInTreePVToCSI` func, so that's safe. I am ok following intree behavior, though in CSI driver, empty equals to `default` namespace. So the code should be like following:

```
		if pv.Spec.ClaimRef != nil  {
			namespace = pv.Spec.ClaimRef.Namespace
		}
```

Reply:
@andyzhangx sounds good - changed.

Reply:
@msau42 could you take a look? thanks.

Reply:
are we ok with this PR? thanks.